runtime/src/accounts_db.rs

```diff
@@ -1255,10 +1317,57 @@ impl AccountsDB {
             .filter(|slot| !accounts_index.is_root(**slot))
             .collect();
         drop(accounts_index);
+        let mut storage_lock_elapsed = Measure::start("storage_lock_elapsed");
         let mut storage = self.storage.write().unwrap();
+        storage_lock_elapsed.stop();
+
+        let mut all_removed_slot_storages = vec![];
+        let mut total_removed_storage_entries = 0;
+        let mut total_removed_bytes = 0;
+
+        let mut remove_storages_elapsed = Measure::start("remove_storages_elapsed");
         for slot in non_roots {
-            storage.0.remove(&slot);
+            let slot_removed_storages = storage.0.remove(&slot);
+            total_removed_storage_entries +=
+                slot_removed_storages.as_ref().map(|x| x.len()).unwrap_or(0);
+            total_removed_bytes += slot_removed_storages
+                .as_ref()
+                .map(|x| x.values().map(|i| i.accounts.capacity()).sum())
+                .unwrap_or(0);
+            all_removed_slot_storages.push(slot_removed_storages);
         }
+        remove_storages_elapsed.stop();
+        drop(storage);
+
+        let num_slots_removed = all_removed_slot_storages.len();
+
+        let mut drop_storage_entries_elapsed = Measure::start("drop_storage_entries_elapsed");
+        // Backing mmaps for removed storages entries explicitly dropped here outside
+        // of any locks
+        drop(all_removed_slot_storages);
+        drop_storage_entries_elapsed.stop();
+
+        datapoint_info!(
```

Comment:
(just a note): I think the frequency of submissions of this new metrics should be safe because the freq. of `::purge_slots()` is bound to the slot rate (i.e. 400ms).